### DISCUSSION TITLE
Best Practice To Set Build Changelog Via CI

### DISCUSSION THREAD

**Discussion by Blackjacx** [fastlane/fastlane#19513] [Q&A]
Hey there 👋

we deploy our builds via Bitrise on a nightly basis. Setting the changelog though increases the runtime of each CI integration by 5-10 minutes since it seems that each build has to be processed by Apple first until the changelog can be set. I have several questions here:

1. Is this still state-of the art behavior or did Apple maybe change something here so we don't have this senseless wait interval. 
2. Maybe this can be set faster using the App Store Connect API?
3. Is this working correctly in Fastlane? I saw that when `changelog` **and** `skip_waiting_for_build_processing=false` are set, Fastlane does only wait until the app appears. But 5-10 minutes still feel very long for that.

Besides of that maybe one of you has a "best practice" of how we could circle around making every build 5-10 minutes longer and update the changelog somehow else.

**Comment by alibstrd** [comment]
Hi there
Did you happen to find the solution to this issue?
